controllers/provisioning_controller.go

```diff
@@ -227,13 +229,32 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	// Create metal3 deployment object based on current baremetal configuration
 	metal3Deployment := provisioning.NewMetal3Deployment(ComponentNamespace, &containerImages, &baremetalConfig.Spec)
 
+	// Apply new deployment or compare against existing deployment and update
 	err = provisioning.ApplyNewOrUpdatedDeployment(r.KubeClient.AppsV1(), ComponentNamespace, metal3Deployment, r.Generations, ComponentName)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to apply Metal3 deployment: %v", err)
+		return ctrl.Result{}, fmt.Errorf("unable to apply metal3 deployment: %v", err)
+	}
+
+	// Determine the status of the deployment
+	deploymentState, err := provisioning.GetDeploymentState(r.KubeClient.AppsV1(), ComponentNamespace, baremetalConfig)
+	if apierrors.IsNotFound(err) {
+		// Deployment progressing
+		return ctrl.Result{RequeueAfter: time.Minute}, nil
```

Comment:
We're the ones creating the Deployment, so if it's not there then someone has deleted it; I don't think there's any reason not to requeue right away so we can create it ASAP. In fact, I'd consider just returning the NotFound error, since we should have *just* created it.

Reply:
I have simplified the error handling in this path. I believe it takes care of all the above comments regarding error handling.